src/tiled/objectselectiontool.cpp

```diff
@@ -551,8 +586,25 @@ void ObjectSelectionTool::mouseReleased(QGraphicsSceneMouseEvent *event)
             break;
         }
         const Qt::KeyboardModifiers modifiers = event->modifiers();
+        QSet<MapObjectItem*> selection = mapScene()->selectedObjectItems();
+        if (modifiers & Qt::AltModifier) {
+            auto underlyingObjects = objectItemsAt(event->scenePos());
+
+            if (underlyingObjects.isEmpty())
+                break;
+            auto currRotation = std::find(underlyingObjects.begin(),
+                                          underlyingObjects.end(), mSelectedInUnderlyingRotation);
+            if (currRotation == underlyingObjects.end())
+                currRotation = underlyingObjects.begin();//new stack of objects
+            else {
+                if ((modifiers & (Qt::ShiftModifier | Qt::ControlModifier)) == 0)
+                    selection.remove(mSelectedInUnderlyingRotation);//deselect previous
```

Comment:
Instead of doing this, please suppress the "Clicking one of the selected items changes the edit mode" below when Alt is held. Unless I'm missing something, that should do the same in less code.